games/TicTacToe/contract/src/main.sw

```diff
@@ -0,0 +1,98 @@
+contract;
+
+use core::*;
+use std::{address::Address, chain::auth::msg_sender, hash::sha256, storage::{StorageMap, get, store}};
+
+use data_structures::{Game, Players, Winner, core::ops::Eq};
+
+storage {
+    game: Game,
+    player_turn: u64,
+    map: StorageMap<u64,
+    u64>, 
+}
+
+abi TicTacToe {
```

Comment:
Which of these functions need to be in the ABI? Some of them look like internal functions that should only be called by other functions of this contract.